src/index.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Initialization.
+ */
+import {Router} from './server';
+export const server = new Router();
+
+import {databases} from './database';
+
+console.log(`Server listening on ${server.port}`);
+
+process.on('uncaughtException', (err: Error) => {
+	Router.crashlog(err, 'The main process');
+});
+
+process.on('unhandledRejection', (err: Error) => {
+	Router.crashlog(err, 'A main process promise');
+});
+
+// graceful shutdown.
+process.on('SIGINT', () => {
+	void server.close().then(() => {
```

Comment:
change this sort of thing to async/await

Reply:
I don't think this can be async/await? In my testing it didn't seem to like it.

Reply:
```
// graceful shutdown.
process.on('SIGINT', async () => {
	await server.close()

	// we are no longer accepting requests and all requests have been handled.
	// now it's safe to close DBs
	for (const database of databases) {
		database.close();
	}
});
```
should be equivalent, if it doesn't work then let me know the error

Reply:
Done.